File: frontend/src/lib/types/cdp.ts
export interface CDP {
  numero: number;
  fecha: string;
  codigo_rubro: string;
  cuenta?: string | null;
  objeto: string;
  valor: number;
  estado: string;
  fuente_sifse: number;
  item_sifse: number;
  saldo?: number | null;
}

export interface CDPCreate {
  codigo_rubro: string;
  objeto: string;
  valor: number;
  fuente_sifse?: number;
}

export interface CDPUpdate {
  valor?: number;
  objeto?: string;
  fuente_sifse?: number;
  item_sifse?: number;
}

File: backend/app/schemas/cdp.py
from pydantic import BaseModel, Field


class CDPCreate(BaseModel):
    codigo_rubro: str
    objeto: str
    valor: float = Field(gt=0)
    fuente_sifse: int = 0
    item_sifse: int = 0


class CDPUpdate(BaseModel):
    valor: float | None = Field(default=None, gt=0)
    objeto: str | None = None
    fuente_sifse: int | None = None
    item_sifse: int | None = None


class CDPResponse(BaseModel):
    numero: int
    fecha: str
    codigo_rubro: str
    cuenta: str | None = None
    objeto: str
    valor: float
    estado: str
    fuente_sifse: int
    item_sifse: int
    saldo: float | None = None

    model_config = {"from_attributes": True}
